ios-app/YouPlay/Services/PlaylistService/PlaylistServiceImpl/PlaylistServiceImpl.swift

```diff
@@ -148,6 +148,40 @@ class PlaylistServiceImpl: PlaylistService {
         
         return playlistIdToSongs
     }
+
+    func addSongToLikedSongs(uid: String, song: Song) async {
+        let playlistRef = FirestoreConstants.PlaylistsCollection(uid: uid)
+        
+        do {
+            let querySnapshot = try await playlistRef.whereField("title", isEqualTo: "Liked Songs").getDocuments()
```

Comment:
There a lot of duplicate code between this and just adding a song to a playlist functionality. I think the way to go is just find the ID of the playlist named "Liked songs" (make sure capitalization is ignored) and then just add a song to that playlistId